server.go

```diff
@@ -0,0 +1,75 @@
+package unixtransport
```

Comment:
Should this be a separate package, e.g. `unixtransport/unixtransporttest`? Honest question, don't actually know 😅

Reply:
Up to you. Personally I think that since both the existing and new APIs are so small, it's probably fine to keep it all in one package.

That said I have no qualms with choosing to revisit this later and won't mind updating my code if we decide to move this.